src/persist/disk.rs

```diff
@@ -59,8 +64,23 @@ impl DiskBackend {
             data_path,
             index,
             data_ofs,
+            temp_dir: None,
         })
     }
+
+    fn register_temp_dir(&mut self, dir: TempDir) {
+        self.temp_dir = Some(dir)
+    }
```

Comment:
Unnecessary function call, you could `inline` it but since it's private and only used in one place I would suggest to remove the function and have below just `self.temp_dir = Some(dir);` instead of `db.register_temp_dir(dir);`

Reply:
sure